features/steps/steps.py

```diff
@@ -36,11 +36,38 @@ def when_i_run_command(context, command, user_spec):
     context.process = process
 
 
+@when('I attach "{token}" {user_spec}')
+def when_i_attach_token(context, token, user_spec):
```

Comment:
Either we should make use of the token parameter being passed into this function, or we probably could drop it.

alternative 1:
```
@when('I attach "{token_name}" {user_spec}')
def when_i_attach_token(context, token_name, user_spec):
    token = getattr(context.config, token_name, None)
```
The above would require that you change your  attach_validtoken.feature
```When I attach "<contract_token>" with sudo```
changed to 
```When I attach "contract_token" with sudo```

Reply:
Just touching this point to check to see if it was missed.

Reply:
sorry. Please check if it is right now. I've changed it